src/battle_util.c

```diff
@@ -4977,46 +4977,76 @@ u32 AbilityBattleEffects(u32 caseID, u32 battler, u32 ability, u32 special, u32
     case ABILITYEFFECT_MOVES_BLOCK: // 2
         {
             u16 moveTarget = GetBattlerMoveTargetType(battler, move);
-            u16 battlerAbility = GetBattlerAbility(battler);
-            u16 targetAbility = GetBattlerAbility(gBattlerTarget);
 
-            if ((gLastUsedAbility == ABILITY_SOUNDPROOF && gMovesInfo[move].soundMove && !(moveTarget & MOVE_TARGET_USER))
-             || (gLastUsedAbility == ABILITY_BULLETPROOF && gMovesInfo[move].ballisticMove))
+            switch (gLastUsedAbility)
+            {
+            case ABILITY_SOUNDPROOF:
+                if (gMovesInfo[move].soundMove && !(moveTarget & MOVE_TARGET_USER))
+                    effect = 1;
```

Comment:
Should/can we get an enum to describe what precisely these effect numbers stand for? It's not very intuitive

Reply:
Probably not feasible. They are used all across `AbilityBattleEffects` and can mean anything.

Reply:
Then I think we should streamline it. Out of scope for this PR of course, but it just makes it unnecessarily opaque to see what actually happens